### ISSUE TITLE
6.0.8 breaks typing of custom props with React Native

### ISSUE THREAD

**Issue by pmk1c** [styled-components/styled-components#4165]
## System:
 - OS: macOS 13.5.2
 - CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
 - Memory: 449.48 MB / 16.00 GB
 - Shell: 5.9 - /bin/zsh
## Binaries:
 - Node: 18.16.1 - ~/.asdf/installs/nodejs/18.16.1/bin/node
 - npm: 9.5.1 - ~/.asdf/plugins/nodejs/shims/npm
 - pnpm: 8.7.4 - ~/.asdf/installs/nodejs/18.16.1/bin/pnpm
 - Watchman: 2023.08.28.00 - /usr/local/bin/watchman
## npmPackages:
 - styled-components: ^6.0.7 => 6.0.8 

## Reproduction
```tsx
import React from "react";
import styled from "styled-components/native";

interface Props {
  customProp?: string;
  secondProp?: string;
}

const StyledView = styled.View<Props>`
  ${(props) => {
    // correctly typed
    props.customProp;

    // correctly typed
    props.secondProp;

    return null;
  }}
`;

const Component = () => (
  <StyledView
    // No overload matches this call.
    // Property 'customProp' does not exist on type
    // 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'.
    customProp="test"
  />
);
```

## Expected Behavior
Using custom props with `styled-components/native` should not give a TypeScript error.

## Additional Info
Only the first custom prop of a component gives this error. The others do not.
When doing the same with `styled-components` (non-native) and a `div` instead of a `View` it works without a problem.

**Comment by LucasPMM** [comment]
Same here, using expo 49

**Comment by vm-agnislav** [comment]
The same. Reverted back to 6.0.7

**Comment by centavosx** [comment]
I'm having the same problem with the latest version too.

**Comment by cesarhilario** [comment]
I am using patch-package with this diff file https://gist.github.com/cam-shaw/fb8bb64870bb6889827491f235c01a63

**Comment by krudos** [comment]
try 6.1.0

**Comment by ademarsj** [comment]
Working on 6.1.0, thanks.
